cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/offer/PersistentOfferRequirementProvider.java

```diff
@@ -70,16 +72,42 @@ public OfferRequirement getNewOfferRequirement(CassandraContainer container) {
         }
     }
 
-    public OfferRequirement getReplacementOfferRequirement(Protos.TaskInfo taskInfo) {
-        LOGGER.info("Getting replacement requirement for task: {}", taskInfo.getTaskId().getValue());
-        return getExistingOfferRequirement(taskInfo);
+    public OfferRequirement getReplacementOfferRequirement(CassandraContainer container) {
+        LOGGER.info("Getting replacement requirement for task: {}", container.getId());
+        try {
+            return new OfferRequirement(
+                    clearTaskIds(container.getTaskInfos()),
+                    clearExecutorId(container.getExecutorInfo()),
+                    null,
+                    null);
+        } catch (InvalidRequirementException e) {
+            LOGGER.error("Failed to construct OfferRequirement with Exception: ", e);
+            return null;
+        }
     }
 
     public OfferRequirement getUpdateOfferRequirement(Protos.TaskInfo taskInfo) {
         LOGGER.info("Getting updated requirement for task: {}", taskInfo.getTaskId().getValue());
         return getExistingOfferRequirement(taskInfo);
     }
 
+    private List<Protos.TaskInfo> clearTaskIds(Collection<Protos.TaskInfo> taskInfos) {
+        List<Protos.TaskInfo> outTaskInfos = new ArrayList<>();
+        for (Protos.TaskInfo restartTaskInfo : taskInfos) {
+            outTaskInfos.add(
+                    Protos.TaskInfo.newBuilder(restartTaskInfo)
+                            .setTaskId(Protos.TaskID.newBuilder().setValue(""))
```

Comment:
Doing `clearXXX` gives us an advantage of using `hasXxx` which is less error prone and cleaner than doing a `getXxx` (which will always return a valid object in ProtoBuf world) and then checking whether it's empty. 
Clearing does guarantee that the underlying objects goes away, so effectively `hasXxx` will return false, post `clearXxx` operation. Also, doing an accidental `getXxx` (assuming it's a string), followed by an empty check will also be successful.

Either ways, I'm okay.